tst/EnergyPlus/unit/WaterThermalTanks.unit.cc

```diff
@@ -0,0 +1,77 @@
+// EnergyPlus::WaterThermalTank Unit Tests
+#include <cmath>
+
+// Google Test Headers
+#include <gtest/gtest.h>
+
+// EnergyPlus Headers
+#include <WaterThermalTanks.hh>
+
+using namespace EnergyPlus;
+
+bool FloatEqualTest(Real64 val1, Real64 val2, Real64 SmallNum = 0.00000001) {
+	using namespace std;
+	return ( abs(val1 - val2) < SmallNum );
+}
+
+TEST( HeatPumpWaterHeaterTests, TestQsourceCalcs )
+{
+	Real64 DeltaT = 0.0;
+	Real64 const SourceInletTemp = 62.0;
+	Real64 const Cp = 4178.; // water, J/(kg * K)
+	Real64 const SetPointTemp = 60.0;
+	Real64 const SourceMassFlowRateOrig = 0.378529822165; // water, 6 gal/min
+	Real64 SourceMassFlowRate = SourceMassFlowRateOrig;
+	Real64 Qheatpump = 0.0;
+	Real64 Qsource = 0.0;
+	
+	// Mixed Tank
+	
+	// Test case without HPWH
+	WaterThermalTanks::CalcMixedTankSourceSideHeatTransferRate(DeltaT, SourceInletTemp, Cp, SetPointTemp, SourceMassFlowRate, Qheatpump, Qsource);
+	// Qsource is non zero and calculated relative to the tank setpoint.
+	EXPECT_TRUE(FloatEqualTest(SourceMassFlowRate * Cp * (SourceInletTemp - SetPointTemp), Qsource));
```

Comment:
That's a great suggestion. I didn't see that in the googletest docs. I'll make that change.